src/taipy/core/cycle/_cycle_repository_factory.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2022 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+from typing import Type
+
+from .._repository._repository_factory import _RepositoryFactory
+from ..common._utils import _load_fct
+from ._cycle_fs_repository import _CycleFSRepository
+from ._cycle_repository import _CycleRepository
+
+
+class _CycleRepositoryFactory(_RepositoryFactory):
+    @classmethod
+    def _build_repository(cls) -> Type[_CycleRepository]:  # type: ignore
+        if cls._using_enterprise():
+            return _load_fct(cls._TAIPY_ENTERPRISE_CORE_MODULE + ".cycle._cycle_repository", "_CycleRepository")  # type: ignore
```

Comment:
I guess this will be a method since it will return a different implementation depending on the config (_CycleFSRepository, _CycleSQLRepository, _CycleMongoRepository).
```suggestion
            return _load_fct(cls._TAIPY_ENTERPRISE_CORE_MODULE + ".cycle._cycle_repository", "_CycleRepository")()  # type: ignore
```